### ISSUE TITLE
Three-body Jastrow is not supported with the momentum distribution

### ISSUE THREAD

**Issue by kylanpaa** [QMCPACK/qmcpack#464]
Runs with three-body Jastrow and the momentum distribution estimator result in "Fatal Error. Aborting at OrbitalBase::get_ratios is not implemented by OrbitalBase".

Inputs and outputs are given in the zip-file. The file pwscf.pwscf.h5 can be either recreated with the scf.in or copied from eos (location in the readme.txt).

[J3_nofk_issue.zip](https://github.com/QMCPACK/qmcpack/files/1442087/J3_nofk_issue.zip)

**Comment by prckent** [comment]
Is anyone already familiar with how this functionality differs from the already implemented functionality? One part of the difference should be that for this kind of estimator we need ratios with a least one set of coordinates that are not electron positions, "virtual particles".

**Comment by jngkim** [comment]
It uses 0-th electron as an active electron and compute ratios.
All the elements are in place and just have to be put together.

**Comment by ye-luo** [comment]
@prckent You might noticed that there is a initial VirtualParticleSet implementation but not connected. I have reworked it in order to be used by NLPP evaluation and estimator needs. For the moment, I only cleaned up the class for minimal creation cost. The computation part is not started yet and it will not appear soon, at least after new year and probably I will target SoA first. For the current needs, the optimal way is just implement the missing function get_ratios with minimal effort.
